internal/cmd/egctl/testdata/translate/out/default-resources.all.yaml

```diff
@@ -1082,6 +1082,13 @@ xds:
             routes:
             - match:
                 prefix: /
+              metadata:
+                filterMetadata:
+                  io.envoyproxy.gateway.metadata:
```

Comment:
Would we need to add other metadata in the future? If so, we might need another level within the metadata structure to accommodate them.

Reply:
I changed the field to be a list of resources for future proofing.

Reply:
Actually, I'm having second thoughts about a list here. It's not so easy to address dynamic metadata in a list. So, I think that I'll go with your approach, but maybe slightly generalize to `Gateway`, `Route` and `Backend` top-level resources, which can then be GW/GWC, HTTP/GRPCRoute, Service/ServiceImport/Backend.

Reply:
I just thought the metadata structure should be able to accommodate multiple resources, but I don't know how exactly the structure should look like. Let's brainstorm in the next meeting?